src/nvim/option_defs.h

```diff
@@ -645,6 +645,7 @@ EXTERN char_u *p_titleold;      ///< 'titleold'
 EXTERN char_u *p_titlestring;   ///< 'titlestring'
 EXTERN char_u *p_tsr;           ///< 'thesaurus'
 EXTERN int p_tgc;               ///< 'termguicolors'
+EXTERN char_u *p_twt;           ///< 'termwintype'
```

Comment:
why do we need an option? We can just enable it always, depending on the detected Windows version.

nit: Nvim doesn't have a "terminal window" so it would be unfortunate if we had to copy this Vim option.

Reply:
In the vim manual:

> Windows 10 October 2018	Update is the first version that supports ConPTY, however it is still considered unstable.  ConPTY might become stable in the next release of Windows 10.

Therefore, I think that it is possible to switch with an option.

Reply:
It's not worth the extra code/complication. And the option has a name that doesn't fit with Neovim, since there is no "terminal window" in Neovim.

Reply:
> It's not worth the extra code/complication.

Even the [Windows Terminal](https://github.com/microsoft/terminal) created by Microsoft has a preview status. If there is a problem with `ConPTY`, neovim may not be able to fix it. Besides, you can use a mouse with `winpty`, but you cannot use a mouse with `ConPTY` yet. From these, I think it is better to allow the user to select the backend.

Reply:
I'm really not in favor of this. If it turns out to be a serious problem, we can revisit it later. 

The "preview" status of conpty is temporary. If it has any problems, they will go away as time goes on. Whereas the extra code and documentation we would add for the extra option, won't go away.

Reply:
Terminal tests depend on winpty implementation. They fail in the CI if using conpty.

https://github.com/neovim/neovim/pull/13177